### PR TITLE
[FEATURE] Add `module_alias` resolution capabilities - round 2

### PR DESCRIPTION
**Pull request by davidroeca**
This PR picks up the work done by @bradennapier in https://github.com/facebook/flow/pull/7185, rebasing his work to the master branch and adding a few changes for to support `<PROJECT_ROOT>`.

Still to do:
* [x] Additional tests requested by @samwgoldman [here](https://github.com/facebook/flow/pull/7185#pullrequestreview-201279048)
* [x] Additional tests for `module.system.node.resolve_dirname` if this touches code related to that
* [ ] ~Handle the relative paths from the initial PR~ I'm almost inclined to not tackle this one, since `<PROJECT_ROOT>` will be handled. Thoughts?
* [ ] Final review from @nmote 

Note that this is the first time I've gone through this codebase and also my first time writing OCaml. Please nitpick away! Not sure if I'm following all of the right style conventions.

### CODE REVIEW

File: src/services/inference/module/module_js.ml [hunk 1]
```diff
@@ -408,23 +411,45 @@ module Node = struct
...
+                lazy_seq
+                  ( Files.node_resolver_dirnames file_options
+                  |> Core_list.map ~f:(fun dirname ->
+                        lazy
+                          (resolve_relative
+                            ~options
+                            ~reader
+                            loc
+                            ?resolution_acc
+                            dir (spf "%s%s%s" dirname Filename.dir_sep r))) )
```
Comment (davidroeca) [inline src/services/inference/module/module_js.ml]: I can implement the same `modified_dir` logic here if it's also preferred to support `<PROJECT_ROOT>` for `module.system.node.resolve_dirname`

**Comment by davidroeca** [comment]
@nmote I added some of the incremental tests that were mentioned in the previous PR and think this is ready for review as long as we're ok with only supporting `<PROJECT_ROOT>/alias` and `alias`

**Comment by davidroeca** [comment]
@nmote anything?

**Comment by nmote** [comment]
Sorry for the delay. I think I'll be able to take a close look at this sometime this week.

**Comment by nmote** [comment]
I'll take a closer look shortly, but first I have a question. @gabelevi just added a config option called `module.system.node.root_relative_dirname` (https://github.com/facebook/flow/commit/d17ed03661d8c0af2450cd91645c9641fdab9d69). See the release notes for v0.111 as well as https://github.com/facebook/flow/issues/8156.

Does this intersect at all with what's been done here?

**Comment by nmote** [comment]
I've looked a little bit closer and @gabelevi's recent work does seem to be pretty similar to this. Could you let me know whether what he's done works for your goals? If not, could you explain what you need from this PR that his work doesn't accomplish, and I'll take another look?

Thanks for your patience on this.

**Comment by davidroeca** [comment]
Looks to me like that PR should handle the main use cases I had in mind

**Comment by nmote** [comment]
If that's the case, should this be closed?

**Comment by davidroeca** [comment]
Yes

**Pull request closed without merging**